DatadogRUM/Sources/RUMContext/AccessibilityReader.swift

```diff
@@ -0,0 +1,273 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import DatadogInternal
+import Foundation
+import UIKit
+
+internal final class AccessibilityReader {
+    @ReadWriteLock
+    var state: Accessibility
+
+    private let notificationCenter: NotificationCenter
+    private let accessibilityValues: AccessibilityValues
+    private var observers: [NSObjectProtocol] = []
+
+    init(notificationCenter: NotificationCenter = .default, accessibilityValues: AccessibilityValues = LiveAccessibilityValues()) {
+        state = Accessibility()
+        self.notificationCenter = notificationCenter
+        self.accessibilityValues = accessibilityValues
+        startObserving()
+        updateState()
+    }
+
+    deinit {
+        stopObserving()
+    }
+
+    func updateState() {
+        DispatchQueue.main.async { [weak self] in
```

Comment:
suggestion/ To better encapsulate this component, we can make this method `private`. Other components shouldn't need to manually trigger a state update because in real usage this is managed by `AccessibilityReader` internally.

To what I see, the reason for this being `internal` might be related to testing. I've left a separate comment with a suggestion on how we can improve this while keeping the method private.